spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -62,6 +62,46 @@ module Mocks
           expect(receiver.foo).to eq(4)
         end
 
+        if RSpec::Support::RubyFeatures.kw_args_supported?
+          binding.eval(<<-RUBY, __FILE__, __LINE__)
+          it 'allows a `do...end` block implementation with keyword args to be provided' do
+            wrapped.to receive(:foo) do |**kwargs|
+              kwargs[:kw]
+            end
+
+            expect(receiver.foo(kw: :arg)).to eq(:arg)
+          end
+
+          it 'allows a `do...end` block implementation with optional keyword args to be provided' do
+            wrapped.to receive(:foo) do |kw: :arg|
+              kw
+            end
+
+            expect(receiver.foo(kw: 1)).to eq(1)
+          end
+
+          it 'allows a `do...end` block implementation with optional keyword args to be provided' do
+            wrapped.to receive(:foo) do |kw: :arg|
+              kw
+            end
+
+            expect(receiver.foo).to eq(:arg)
+          end
+          RUBY
+        end
+
+        if RSpec::Support::RubyFeatures.required_kw_args_supported?
+          binding.eval(<<-RUBY, __FILE__, __LINE__)
+          xit 'allows a `do...end` block implementation with required keyword args' do
```

Comment:
This test fails jruby 9.1 & 9.2 in a way that seems like an actual issue.  

I'm happy to remove it as I'm not even sure that anyone would use a mock method in this way.

I don't have jruby locally but I could try to get it set up to debug further if desired.